server/src/core/odoo.rs

```diff
@@ -367,34 +364,15 @@ impl SyncOdoo {
                 current_count = 0;
                 let symbol = sym.borrow_mut();
                 for (index, dep_set) in symbol.get_all_dependencies(step).iter().enumerate() {
-                    if index == BuildSteps::ARCH as usize {
-                        for dep in dep_set.iter() {
-                            if self.rebuild_arch.contains(&dep) {
-                                current_count += 1;
-                            }
-                        }
-                    }
-                    if index == BuildSteps::ARCH_EVAL as usize {
-                        for dep in dep_set.iter() {
-                            if self.rebuild_arch_eval.contains(&dep) {
-                                current_count += 1;
-                            }
-                        }
-                    }
-                    if index == BuildSteps::ODOO as usize {
-                        for dep in dep_set.iter() {
-                            if self.rebuild_odoo.contains(&dep) {
-                                current_count += 1;
-                            }
-                        }
-                    }
-                    if index == BuildSteps::VALIDATION as usize {
-                        for dep in dep_set.iter() {
-                            if self.rebuild_validation.contains(&dep) {
-                                current_count += 1;
-                            }
-                        }
-                    }
+                    let index_set =  match index {
+                        x if x == BuildSteps::ARCH as usize => &self.rebuild_arch,
+                        x if x == BuildSteps::ARCH_EVAL as usize => &self.rebuild_arch_eval,
+                        x if x == BuildSteps::VALIDATION as usize => &self.rebuild_validation,
+                        _ => continue,
+                    };
+                    dep_set.iter()
```

Comment:
I do not really like the for_each :/
Why not a

```
current_count += dep_set.iter().filter(...).count();
```

Reply:
That looks good too, I will add it